Branch-SDK/Branch-SDK/BNCError.m

```diff
@@ -6,10 +6,49 @@
 //  Copyright (c) 2014 Branch Metrics. All rights reserved.
 //
 
+
 #import "BNCError.h"
 
-NSString * const BNCErrorDomain = @"io.branch";
 
-@implementation BNCError
+NSString * const BNCErrorDomain = @"io.branch.error";
+
+
+#define _countof(arr) (sizeof(arr) / sizeof(arr[0]))
+
+
+NSError *_Nonnull BNCErrorWithCode(BNCErrorCode errorCode) {
+    return BNCErrorWithCodeAndReason(errorCode, nil);
+}
+
+
+NSError *_Nonnull BNCErrorWithCodeAndReason(BNCErrorCode errorCode, NSString* reason) {
+
+    NSString *const localizedDescriptions[] =
```

Comment:
Just curious, is there a reason for choosing a C array instead of `NSArray`? There will be no need for using the `_countof` macro because you can use the `count` built-in method